workers/data_refinery_workers/processors/salmon.py

```diff
@@ -285,33 +252,113 @@ def _run_salmon(job_context: Dict, skip_processed=SKIP_PROCESSED) -> Dict:
 
     return job_context
 
-def _run_tximport(job_context: Dict) -> Dict:
-    """ Runs the `tximport` R package to find genes. 
-    
-    For an R code example of tximport usage, see: 
-    https://github.com/jaclyn-taroni/ref-txome/blob/79e2f64ffe6a71c5103a150bd3159efb784cddeb/4-athaliana_tximport.R
 
-    Related: https://github.com/data-refinery/data-refinery/issues/82
+def _run_salmontools(job_context: Dict, skip_processed=SKIP_PROCESSED) -> Dict:
+    """ Run Salmontools to extract unmapped genes. """
 
-    """
+    logger.debug("Running SalmonTools ...")
+    skip = False
+    unmapped_filename = job_context['output_directory'] + 'aux_info/unmapped_names.txt'
+    if skip_processed and os.path.exists(unmapped_filename):
+        logger.info("Skipping pre-processed SalmonTools run!")
+        skip = True
 
-    try:
-        tx_out = ro.r['::']('tximport', 'tximport')(job_context['output_directory'] + 'quant.sf', type="salmon", txOut=True)
-    except RRuntimeError as e:
-        # XXX/TODO: This doesn't work yet.
-        # But, eventually, it will look something like this. ^^
+    job_context['time_start'] = timezone.now()
+    if skip:  # If this procedure should be skipped, return immediately
+        job_context['time_end'] = timezone.now()
```

Comment:
I don't think anything else is going to use this, so if you're not running salmon tools it probably doesn't need to be set.

Reply:
What about `job_context['time_start']` if the job is skipped? Should we delete that one too?

Reply:
If we switch to using local variables it won't be an issue.

Reply:
Good to know. I thought each job would always set the keys such as `time_start` and `time_end` in `job_context`.

Reply:
I can see how it would look like that. The `start_time` and `end_time` fields for the ProcessorJob are set in the utils functions `start_job` and `end_job`.